src/app/navigation/utils.ts

```diff
@@ -109,3 +110,15 @@ export const findPathForParams = (params: object, paths: string[]) => {
       paramsInPath.every(({name}) => paramKeys.includes(name.toString()));
   });
 };
+
+export const getTargetFromQuery = (query: OutputParams): {[key: string]: any} | null => {
```

Comment:
I think it might be against Tom's suggestion from: https://github.com/openstax/rex-web/pull/514#pullrequestreview-435226251
> i don't like having navigation be aware of the specific types of scroll targets. I want the highlight or search logic to be able to tell if the target does or does not apply to them, but having the specific selectors in navigation seems bad to me

but I've already implemented this because I've realized that was part of my another approach after I've done this

Please don't close this conversation so Tom can take a look on it

I think having selectors for `searchScrollTarget` and `highlightScrollTarget` in the `navigation/selectors.ts` would be okay

Reply:
I think there can't be more than one at a given time so i believe that the scroll target should be somewhere, with no distinction if it is a highlight or search scroll target. Determine what kind of a scroll target it is when it needs to be handled.

Reply:
For now, I've created `scrollTarget` selector in `navigation` and just check if this is proper scroll target directly in the logic which needs one of its types